### ISSUE TITLE
Create server_capability response message is incorrect

### ISSUE THREAD

**Issue by ZugNZwang** [apache/trafficcontrol#3999]
## I'm submitting a ...

- [X] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

When HTTP post https://{{TO_BASE_URL}}/api/{{api_version}}/server_capabilities with no request body is made the response alert message received is incorrect 

## Expected / new behavior:

```
{
    "alerts": [
        {
            "text": "missing entries",
            "level": "error"
        }
    ]
}
```
## Minimal reproduction of the problem with instructions:

```
{
   "alerts": [
       {
           "text": "EOF",
           "level": "error"
       }
   ]
}
```
## Anything else:

on latest master - last commit  3.0.0-10254.f1d15647.el7

**Comment by mitchell852** [comment]
does this happen on any other POSTs in our api?

**Comment by rob05c** [comment]
I don't think this is a bug, it's a fundamental change to the API.

For a 404 vs an empty array for objects that don't exist, I don't think either is the "right" solution. A 404 is probably more expected. But to say "the API returns an array of objects, and if your filter didn't have any, you get a successful empty array" is not a violation of HTTP.

Moreover, the TO API has returned 200 [] for ages. It isn't documented behavior, but it's most certainly a specific behavior, that clients almost certainly rely on in places. Even if it doesn't violate the SemVer API Promise, we can't reasonably change this without breaking tons of people.

Likewise, IMO we should continue to make new endpoints follow the same pattern as everything else.

It wouldn't be a bad idea to do this in the next major version of the API (2.0), but we simply can't safely do it to 1.x.

**Comment by rob05c** [comment]
>does this happen on any other POSTs in our api?

Yes, I just tested, looks like it's all endpoints, not just this one. Probably a bug in the TO HTTP dispatch/routing.

**Comment by ocket8888** [comment]
I don't think it's dispatch/routing, that to me looks like the kind of error `encoding/json.Decoder.Decode` would return if the first character in its input buffer is `EOF`. I think requests are probably getting to the right place, and we're just returning exactly the error we got.
